Library/Homebrew/cmd/deps.rb

```diff
@@ -129,41 +186,52 @@ def deps_for_formulae(formulae, recursive = false, &block)
 
   def puts_deps(formulae)
     formulae.each do |f|
-      deps = deps_for_formula(f).sort_by(&:name).map(&method(:dep_display_name))
+      deps = deps_for_formula(f)
+      deps = condense_requirements(deps)
+      deps = deps.sort_by(&:name).map(&method(:dep_display_name))
       puts "#{f.full_name}: #{deps.join(" ")}"
     end
   end
 
-  def puts_deps_tree(formulae)
+  def puts_deps_tree(formulae, recursive = false)
     formulae.each do |f|
-      puts "#{f.full_name} (required dependencies)"
-      recursive_deps_tree(f, "")
+      puts f.full_name
+      @dep_stack = []
+      recursive_deps_tree(f, "", recursive)
       puts
     end
   end
 
-  def recursive_deps_tree(f, prefix)
-    reqs = f.requirements.select(&:default_formula?)
-    deps = f.deps.default
-    max = reqs.length - 1
-    reqs.each_with_index do |req, i|
-      chr = if i == max && deps.empty?
-        "└──"
-      else
-        "├──"
-      end
-      puts prefix + "#{chr} :#{dep_display_name(req.to_dependency)}"
-    end
-    max = deps.length - 1
-    deps.each_with_index do |dep, i|
+  def recursive_deps_tree(f, prefix, recursive)
+    reqs = f.requirements
+    deps = f.deps
+    dependables = reqs + deps
+    dependables = dependables.reject(&:optional?) unless @with_optional
+    dependables = dependables.reject(&:build?) unless @with_build
+    dependables = dependables.reject(&:recommended?) if @skip_recommended
+    max = dependables.length - 1
+    @dep_stack.push f.name
+    dependables.each_with_index do |dep, i|
+      next if !@with_reqs && dep.is_a?(Requirement) && !dep.default_formula?
       chr = if i == max
         "└──"
       else
         "├──"
       end
-      prefix_ext = (i == max) ? "    " : "│   "
-      puts prefix + "#{chr} #{dep_display_name(dep)}"
-      recursive_deps_tree(Formulary.factory(dep.name), prefix + prefix_ext)
+      display_s = "#{chr} #{dep_display_name(dep)}"
```

Comment:
how about `tree_lines`, to distinguish the graphical lines of the tree structure from a "line" of text?